File: week1-python/Basic_Image_Operations/Creating_an_Image_Mask.py
# Import libraries
import cv2
import numpy as np
from dataPath import DATA_PATH

# Read image
image = cv2.imread(DATA_PATH+"images/boy.jpg")

# Let's see what image we are dealing with
#cv2.imshow("Image",image)
#cv2.waitKey(0)

# Create an empty image of same size as the original
mask1 = np.zeros_like(image)
#cv2.imshow("Mask",mask1)
#cv2.waitKey(0)
cv2.imwrite("../results/mask.png",mask1)

mask1[50:200,170:320] = 255
print(mask1.dtype)
#cv2.imshow("Original Image",image)
#cv2.imshow("Mask",mask1)
#cv2.waitKey(0)

mask2 = cv2.inRange(image, (0,0,150), (100,100,255))
#cv2.imshow("Original Image",image)
#cv2.imshow("Masked Image",mask2)
#cv2.waitKey(0)
cv2.imwrite("../results/maskedImage.png",mask2)

File: week1-cpp/Basic_Image_Operations/Creating_an_image_mask.cpp
// Include libraries
#include <iostream>
#include "dataPath.hpp"
#include <opencv2/opencv.hpp>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;

int main(void){
        // Read image
        Mat image = imread(DATA_PATH+"images/boy.jpg");

	// Create an empty image of same size as the original
	Mat mask1 = Mat::zeros(image.size(), image.type());
	imwrite("../results/mask1.png",mask1);
	//imshow("Mask 1",mask1);
	//waitKey(0);

	mask1(Range(50,200),Range(170,320)).setTo(255);
	imwrite("../results/mask1Revised.png",mask1);
	//imshow("Mask 1 Revised",mask1);
	//waitKey(0);

	Mat mask2;
	inRange(image, Scalar(0,0,150), Scalar(100,100,255), mask2);
	imwrite("../results/mask2.png",mask2);
	//imshow("Mask 2",mask2);
	//waitKey(0);

	return 0;
}
